### PR TITLE
proposal: improve postprocessor

### PR DESCRIPTION
**Pull request by jerryjliu**
here we allow postprocessor to take in query_str, not just QueryBundle

also edited docs slightly. idea is to make it less painful to use our postprocessors on their own

### CODE REVIEW

File: llama_index/indices/postprocessor/types.py [hunk 1]
```diff
@@ -33,10 +33,25 @@ def _get_prompt_modules(self) -> PromptMixinType:
     def class_name(cls) -> str:
         return "BaseNodePostprocessor"
 
-    @abstractmethod
     def postprocess_nodes(
         self,
         nodes: List[NodeWithScore],
         query_bundle: Optional[QueryBundle] = None,
+        query_str: Optional[str] = None,
```
Comment (Disiok) [inline llama_index/indices/postprocessor/types.py:40]: nit: should we just use single param, to be consistent with e.g. retriever and query engine:

```
    def retrieve(self, str_or_query_bundle: QueryType) -> List[NodeWithScore]:
```

File: llama_index/param_tuner/base.py [hunk 2]
```diff
@@ -262,13 +262,16 @@ def param_fn_wrapper(
 
         results = tuner.fit()
         all_run_results = []
-        for result in results:
+        for idx in range(len(results)):
+            result = results[idx]
```
Comment (Disiok) [inline llama_index/param_tuner/base.py:266]: what's motivation of this change?

**Review by Disiok — APPROVED** [review]

File: llama_index/param_tuner/base.py (hunk 2 above)
Comment (jerryjliu) [reply llama_index/param_tuner/base.py:266]: oh lol. this was causing a lint error if you install ray tune (ResultGrid is not iterable but it has getitem implemented..)

Comment (Disiok) [reply llama_index/param_tuner/base.py:266]: lol gotcha!

File: llama_index/indices/postprocessor/types.py (hunk 1 above)
Comment (Disiok) [reply llama_index/indices/postprocessor/types.py:40]: oh gotcha, kk I guess this could work.

**Pull request merged**